solutions/usaco-571.mdx

```diff
@@ -2,55 +2,165 @@
 id: usaco-571
 source: USACO Silver 2015 December
 title: High Card Wins
-author: Óscar Garries
+author: Óscar Garries, Ryan Chou
 ---
 
-[Official Analysis](http://www.usaco.org/current/data/sol_highcard_silver_dec15.html)
+[Official Analysis (Java)](http://www.usaco.org/current/data/sol_highcard_silver_dec15.html)
 
-<LanguageSection>
-<CPPSection>
+Intuitively, we always want Bessie to win with the smallest card possible. We can ensure this minimum by storing the available cards for both Bessie and Elsie. If Elsie's current card is higher than Bessie's, we can move to the Bessie's next higher card.
+
+You can see the algorithm in action with the test set below:
+
+<img src = "/usaco-571-anim/highcard_algo.gif"/>
 
 ## Implementation
 
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+
+<CPPSection>
+
 ```cpp
-#include <bits/stdc++.h>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <cstdio>
 
 using namespace std;
 
-int main () {
+// initializes every element to false
+bool elsieHas[100001]{};
```

Comment:
magic number is a bad practice
make a variable like `MAX_CARDS = 1e5` or smth
then put `bool elsieHas[MAX_CARDS + 1]`